.github/workflows/devconfig-validator.yml

```diff
@@ -0,0 +1,219 @@
+name: DevConfig Validator
+
+on:
+  pull_request:
+    types: [synchronize, opened, reopened, edited]
+
+permissions:
+  contents: read
+  pull-requests: write
+  issues: write
+
+jobs:
+  validate-devconfig:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    outputs:
+      needs_devconfig: ${{ steps.check-needs.outputs.needs_devconfig }}
+      has_devconfig: ${{ steps.check-files.outputs.has_devconfig }}
+      devconfig_content: ${{ steps.generate-preview.outputs.devconfig_content }}
+      pr_number: ${{ github.event.pull_request.number }}
+      pr_head_ref: ${{ github.event.pull_request.head.ref }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check for existing DevConfig files
+        id: check-files
+        run: |
+          # Check if .DevConfigs directory exists and has files
+          if [ -d "./generator/.DevConfigs" ] && [ "$(ls -A ./generator/.DevConfigs 2>/dev/null)" ]; then
+            echo "has_devconfig=true" >> $GITHUB_OUTPUT
+            echo "DevConfig files already exist in this PR"
+          else
+            echo "has_devconfig=false" >> $GITHUB_OUTPUT
+            echo "No DevConfig files found in this PR"
+          fi
+
+      - name: Check if DevConfig is needed
```

Comment:
I might be missing something but depending on when the contributor branched from `development` there might be dev config files for other unshipped PRs in the forked branch. Would the workflow say all is good because there is a dev config in the folder even though it is unrelated to what the PR does?